### ISSUE TITLE
Feature: Automatically switch to PiP

### ISSUE THREAD

**Issue by EvilBMP** [mediathekview/zapp#319]
It would be nice, if Zapp would automatically switch to PiP mode, if one minimizes the app or switch to another app, via app switcher.

Also, it would be nice, if PiP is active, that turning off the screen would result in automatic background playback.

So going to the 3-dots menu in the top right corner during playback could be skipped.

YouTube Vanced works/ worked that way and I find that really convenient 🙂

**Comment by cemrich** [comment]
I find this feature very stressfull and would not like to enable it by default. 
But i'll try to find out if this is something more users might like.

**Comment by EvilBMP** [comment]
I have no problem with a switch in the settings, that allows to en- or disable such a feature ;-)

**Comment by Ver-Nunft** [comment]
that would be two great options

**Comment by cemrich** [comment]
Note to self:

This would require a new preferences section along the line of "video player". This may contain the existing "Videos always in landscape mode" setting and the new "Resume playback in picture in picture mode" setting.

**Comment by cemrich** [comment]
# How other apps do it

## Youtube
- back press: mini player
- app switch: stop playback

## New Pipe
- back press: mini player
- app switch: switch to audio playback

## Netflix
- back press: stop playback
- app switch: stop playback

## VLC
- back press: stop playback
- app switch: can be changed:
  1. stop
  1. audio playback
  1. picture in picture

**Comment by cemrich** [comment]
Switching to picture in picture on back press is very easy by placing this in `AbstractPlayerActivity::onCreate`:

```
onBackPressedDispatcher.addCallback(object : OnBackPressedCallback(true) {
	override fun handleOnBackPressed() {
		MultiWindowHelper.enterPictureInPictureMode(this@AbstractPlayerActivity)
	}
})
```

However, there are some subtle bugs when switching videos while in picture in picture mode. And there is #168

**Comment by cemrich** [comment]
Switching to picture in picture on back press is very easy by placing this in `AbstractPlayerActivity::onCreate`:

```
onBackPressedDispatcher.addCallback(object : OnBackPressedCallback(true) {
	override fun handleOnBackPressed() {
		MultiWindowHelper.enterPictureInPictureMode(this@AbstractPlayerActivity)
	}
})
```

However, there are some subtle bugs when switching videos while in picture in picture mode. And there is #168

**Comment by cemrich** [comment]
The related bugs first to fix are: #341 #340

**Comment by cemrich** [comment]
This issue is now implemented with upcoming version 8.1.0. Pip will start (after enabling it in preferences) after a back press from the video player. Turning the screen off during playback will switch to background playback.
